File: bin/Nodes/C/Distance.c
#include "Imogen.h"

typedef struct Distance_t
{
	int PassCount;
} Distance;

int main(Distance *param, Evaluation *evaluation, void *context)
{
    int imgWidth, imgHeight;
	int source = evaluation->inputIndices[0];
	if (source != -1 && GetEvaluationSize(context, source, &imgWidth, &imgHeight) == EVAL_OK)
    {
        param->PassCount = log2(imgWidth)+3;
    }    
    return EVAL_OK;
}

File: bin/Nodes/JS/Distance.js
function Distance(parameters, evaluation, context)
{
	const source = evaluation.inputIndices[0];
	const res = GetEvaluationSize(context, source);
	if (source != -1 && res[2] == EVAL_OK)
    {
        SetParameter(context, evaluation.targetIndex, "passCount", Math.log2(res[0])+1);
    }    
    return EVAL_OK;
}